sdk/src/beta9/runner/endpoint.py

```diff
@@ -110,18 +118,23 @@ async def task_lifecycle(request: Request):
             status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Failed to start task"
         )
 
+    task_lifecycle_data = TaskLifecycleData(
+        status=TaskStatus.Complete,
+        result=None,
```

Comment:
won't this result always be None? what's the point of adding this dataclass wrapper?

Reply:
the result can change. SImilar to how the task_status previously updated through the `yield`, we pass `task_lifecycle_data` by reference into the `api request` where it will update its values. Once this function gets passed the yield, the result is populated